bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/core/listeners/LimitsListener.java

```diff
@@ -38,6 +63,17 @@ public void propertyChange(PropertyChangeEvent evt) {
 		MarkerManager mm = v.getSchedule().getMarkerManager();
 		mm.clearMarkers(this, v);
 
+		final TimeZone zone = v.getSchedule().getSite().timezone();
+
+		final Site site = v.getSchedule().getSite();
+
+		final SortedSet<Block> blocks = v.getSchedule().getBlocks();
+		Interval nnight = null;
+		if (!blocks.isEmpty()) {
+			nnight = blocks.first().getInterval();
+		}
+		final Option<Interval> night = ImOption.apply(nnight);
```

Comment:
If you wanted, you could avoid the `nnight` with the `? :` operator thingy:

```
final Option<Interval> night = ImOption.apply(blocks.isEmpty() ? null : blocks.first().getInterval());
```

This seems fine though.